lib/pages/trade/DesktopTrade.dart

```diff
@@ -602,8 +610,15 @@ class _DesktopTradeState extends State<DesktopTrade> {
                       ],
                     ),
                   ),
-                  ApproveButton(175, 40, "Approve", swapController.approve,
-                      swapController.swap, transactionConfirmed),
+                  ApproveButton(
+                      175,
+                      40,
+                      "Approve",
+                      swapController.approve,
+                      swapController.swap,
+                      canTrade == true
+                          ? transactionConfirmed
+                          : transactionConfirmed2),
```

Comment:
The ticket specifies that the issue is in the buy dialog and not in the trade page